benchmark/test/test_driver.py

```diff
@@ -0,0 +1,152 @@
+from itertools import count, repeat
+
+from zope.interface import implementer
+
+from twisted.internet.defer import Deferred, succeed, fail
+from twisted.trial.unittest import TestCase
+
+from benchmark.benchmark_driver import benchmark
+from benchmark.benchmark_operations import IProbe, IOperation
+
+
+class FakeMeasurement:
+
+    def __init__(self, measurements):
+        """
+        :param measurements: An iterator providing measurement to be
+            returned on each call.
+        """
+        self.measurements = measurements
+
+    def __call__(self, f, *a, **kw):
+        def finished(ignored):
+            return next(self.measurements)
+
+        d = f(*a, **kw)
+        d.addCallback(finished)
+        return d
```

Comment:
Maybe this should go into `measurements` itself, and become a verified fake? :grey_question: